### PR TITLE
Remove optin field from the employee creation form

### LINKED ISSUE #15792: Remove a field from the employee creation form
**Describe the bug**
A little bit of cleanup in the fields of the employee creation form. Please note that if the field must disappear, the associated behavior must be removed too.

**Additional information**
PrestaShop version: 1.7.x

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Removes a field from the employee form
| Type?         | improvement
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Fixes #15792
| How to test?  | Check that the field is not in the form anymore.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15798)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 1]
```diff
@@ -12,3 +12,6 @@ ALTER TABLE `PREFIX_supply_order_detail` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_stock` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_product_attribute` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_product` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
+
+/* Set optin field value to 0 in employee table */
+UPDATE `PREFIX_employee` SET `optin` = 0;
```
Comment (PierreRambaud) [inline install-dev/upgrade/sql/1.7.7.0.sql]: You also need to change the `NOT NULL` or add the default value to 0 to be consistent

Comment (Progi1984) [reply install-dev/upgrade/sql/1.7.7.0.sql]: The same !

Comment (matthieu-rolland) [reply install-dev/upgrade/sql/1.7.7.0.sql]: Actually I removed the column from `install-dev/data/db_structure.sql` , we decided to let the column for people who upgrade (but set the value to zero for all), but to remove it completely for new installations.

Comment (matthieu-rolland) [reply install-dev/upgrade/sql/1.7.7.0.sql]: You guys are right, I missed your point, new added elements would have the default value for this field, which is 1.

Comment (matthieu-rolland) [reply install-dev/upgrade/sql/1.7.7.0.sql]: done !

File: install-dev/upgrade/sql/1.7.7.0.sql [hunk 2]
```diff
@@ -12,3 +12,7 @@ ALTER TABLE `PREFIX_supply_order_detail` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_stock` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_product_attribute` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
 ALTER TABLE `PREFIX_product` ADD `mpn` VARCHAR(40) NULL AFTER `upc`;
+
+/* Set optin field value to 0 in employee table */
+ALTER TABLE `PREFIX_employee` MODIFY COLUMN `optin` tinyint(1) unsigned NOT NULL DEFAULT 0;
```
Comment (PierreRambaud) [inline install-dev/upgrade/sql/1.7.7.0.sql]:
```suggestion
ALTER TABLE `PREFIX_employee` MODIFY COLUMN `optin` tinyint(1) unsigned DEFAULT NULL;
```
At least the developers can save old values. (Which newer having a null value)
Or it's mayb useless

Comment (matthieu-rolland) [reply install-dev/upgrade/sql/1.7.7.0.sql]: @PierreRambaud  I think we don't want those values to be saved.

Comment (Progi1984) [reply install-dev/upgrade/sql/1.7.7.0.sql]: But if we apply the patch of the PR, no value can't be saved.

File: install-dev/data/db_structure.sql [hunk 3]
```diff
@@ -822,7 +822,6 @@ CREATE TABLE `PREFIX_employee` (
   `bo_width` int(10) unsigned NOT NULL DEFAULT '0',
   `bo_menu` tinyint(1) NOT NULL DEFAULT '1',
   `active` tinyint(1) unsigned NOT NULL DEFAULT '0',
-  `optin` tinyint(1) unsigned NOT NULL DEFAULT '1',
```
Comment (PierreRambaud) [inline install-dev/data/db_structure.sql]: Should be removed :)

Comment (PierreRambaud) [inline install-dev/data/db_structure.sql:825]: Should not be removed if we keep the optin for module compliancy.

Comment (PierreRambaud) [reply install-dev/data/db_structure.sql:825]: Look like your remove `id_last_order` field :sweat_smile:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
You removed `id_last_order` instead of adding the `optin` field :D

**Review by PierreRambaud — DISMISSED** [review]

**Review by jolelievre — COMMENTED** [review]
@matthieu-rolland  Some conflicts on the sql file, but the rest is legit

**Review by PierreRambaud — DISMISSED** [review]
> Reviewed 12 of 14 files at r1, 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15798)* status: all files reviewed, 1 unresolved discussion (waiting on @matthieu-rolland, @Progi1984, and @sarahdib)

___
*[install-dev/upgrade/sql/1.7.7.0.sql, line 17 at r2](https://reviewable.io/reviews/prestashop/prestashop/15798#-LqCFuaS0QOGnhFbDTrT-r2-17:-LrDguLG4Doy5QgF-8DG:b-aeuq6d) ([raw file](https://github.com/prestashop/prestashop/blob/dd5cea44df01f0314d2db938bcd240d345113363/install-dev/upgrade/sql/1.7.7.0.sql#L17)):*
<details><summary><i>Previously, Progi1984 (Progi1984) wrote…</i></summary><blockquote>

But if we apply the patch of the PR, no value can't be saved.
</blockquote></details>

We maybe don't want, but if modules does we must keep the behavior :)

**Review by matks — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
@sarahdib FYI, this PR doesn't remove the optin column in database, and it doesn't change the existing values inside it.

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15798)* status: 13 of 14 files reviewed, 1 unresolved discussion (waiting on @PierreRambaud and @sarahdib)

___
*[install-dev/upgrade/sql/1.7.7.0.sql, line 17 at r2](https://reviewable.io/reviews/prestashop/prestashop/15798#-LqCFuaS0QOGnhFbDTrT-r2-17:-LrIjIVFC8S2LGQgKTxX:bs7wumb) ([raw file](https://github.com/prestashop/prestashop/blob/dd5cea44df01f0314d2db938bcd240d345113363/install-dev/upgrade/sql/1.7.7.0.sql#L17)):*
<details><summary><i>Previously, PierreRambaud (GoT) wrote…</i></summary><blockquote>

We maybe don't want, but if modules does we must keep the behavior :)
</blockquote></details>

Done.

**Pull request merged**